### PR TITLE
[ML-Dataframe] add stop and delete endpoints

### PR DESCRIPTION
**Pull request by hendrikmuhs**
**FEATURE BRANCH PR**

add rest endpoints to stop and delete feature build indexer tasks and fix task saving/reloading

### CODE REVIEW

File: x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportDeleteFeatureIndexBuilderJobAction.java
Comment (droberts195) [inline x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportDeleteFeatureIndexBuilderJobAction.java]: This has changed on master.  `threadpool` and `indexNameExpressionResolver` are no longer passed to `super`.  When you are in between PRs it's probably a good idea to merge master into your feature branch and resolve these differences.

File: x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobState.java
Comment (droberts195) [inline x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobState.java]: If the intention is for this to be immutable then you should wrap the `new TreeMap` with `Collections.unmodifiableSortedMap()`, because otherwise a caller can modify it via the getter.

File: x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobState.java
Comment (droberts195) [inline x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/FeatureIndexBuilderJobState.java]: Similar to above, `new TreeMap` should be wrapped with `Collections.unmodifiableSortedMap()`.

File: x-pack/plugin/ml-feature-index-builder/src/test/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/StopFeatureIndexBuilderJobActionRequestTests.java
Comment (droberts195) [inline x-pack/plugin/ml-feature-index-builder/src/test/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/StopFeatureIndexBuilderJobActionRequestTests.java]: I think it should extend `AbstractStreamableTestCase` instead of `AbstractXContentTestCase`.  As it stands at the moment this test is not testing that the wire streaming read/write methods are consistent.

**Review by droberts195 — COMMENTED** [review]
Looks good apart from a few minor nits

File: x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportDeleteFeatureIndexBuilderJobAction.java
Comment (hendrikmuhs) [reply x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/action/TransportDeleteFeatureIndexBuilderJobAction.java]: :+1: will merge master and repair the code after merging this PR

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**
